### ISSUE TITLE
Compare label values of a single metric

### ISSUE THREAD

**Issue by HardikSo** [prometheus/prometheus#11345]
## Proposal
**Use case. Why is this important?**
Some exporters give important information in form of labels in a single metric.
For example, mysqld exporter has a metric 
`mysql_global_status_wsrep_cluster_info{"cluster-uuid":"1122", "local-uuid":"1122"}`
If the cluster uuid is not equal to local uuid, an alert needs to be raised as all the nodes in the cluster needs to have the same cluster uuid.
I have not found any functionality regarding comparing labels of a single metric.

**Comment by kaffarell** [comment]
I don't think this is currently possible in prometheus. If you are using grafana though, you can realize this using [variables](https://grafana.com/docs/grafana/latest/dashboards/variables/).

**Comment by IldarMinaev** [comment]
Try something like this:
mysql_global_status_wsrep_cluster_info unless ignoring(__dst__) (label_replace(mysql_global_status_wsrep_cluster_info, "__dst__", "$1", "cluster-uuid", "(.*)") == label_replace(mysql_global_status_wsrep_cluster_info, "__dst__", "$1", "local-uuid", "(.*)"))

**Comment by erickalmeidaptc** [comment]
Hi @IldarMinaev, your answer was very useful. Thanks.

Now I need to create a chart from this result, something like this: 
`
mysql_global_status_wsrep_cluster_info[1h:1d] unless ignoring(__dst__) (label_replace(mysql_global_status_wsrep_cluster_info, "__dst__", "$1", "cluster-uuid", "(.*)") == label_replace(mysql_global_status_wsrep_cluster_info, "__dst__", "$1", "local-uuid", "(.*)"))
`
But I'm getting this error: `Error executing query: 1:1: parse error: binary expression must contain only scalar and instant vector types`

**Comment by alexgreenbank** [comment]
Hello from the bug scrub.

Apologies for the delay in responding to this.

This is more of a usage question and so converting to a discussion.
